node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/ParkedExpirer.java

```diff
@@ -41,13 +41,14 @@ protected double maintain() {
         NodeList parkedHosts = nodeRepository.nodes()
                                              .list(Node.State.parked)
                                              .nodeType(NodeType.host)
+                                             .not().matching(this::parkedByOperator)
                                              .not().deprovisioning();
         int hostsToExpire = Math.max(0, parkedHosts.size() - MAX_ALLOWED_PARKED_HOSTS);
         parkedHosts.sortedBy(Comparator.comparing(this::parkedAt))
                    .first(hostsToExpire)
                    .forEach(host -> {
                        log.info("Allowed number of parked nodes exceeded. Recycling " + host.hostname());
-                       nodeRepository.nodes().deallocate(host, Agent.ParkedExpirer, "Expired by ParkedExpirer");
+                       nodeRepository.nodes().deprovision(host.hostname(), Agent.ParkedExpirer, Instant.now());
```

Comment:
This just changes the node state, need to actually make a call to AWS to remove the instance, see https://github.com/vespa-engine/vespa/blob/8ef499e16e9fb5daede071d36cb523f4d30538c0/node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/DynamicProvisioningMaintainer.java#L134:L135

Reply:
Won't `DynamicProvisioningMaintainer` make the call to AWS, as the hosts are set to `wantToDeprovision` here?

Reply:
Seems to me like this is equal to setting the host to `dirty`, both of these just modify the second condition for DPM to deprovision and neither actually helps if there are allocated children?

Reply:
Do we want to expire non-empty hosts at all though? If I remember correctly we're adding this to handle hosts expired by ProvisionedExpirer (which will always be empty).

Reply:
In that case, do we even need this maintainer? Cant ProvisionedExpirer just delete them directly?

Reply:
True, ProvisionedExpirer could do it instead. Park as it does today, but mark excess hosts wantToDeprovision could work.

Reply:
The reason it became a separate maintainer is we wanted to remove the oldest parked hosts, but I suppose that's not really required. If anything, it could be an annoyance that we continuously rotate the set of parked hosts.

Reply:
Yes. Let's just keep the oldest ones for now. So basically:
1. ProvisionedExpirer parks hosts that have been too long in provisioned (same as today)
2. ProvisionedExpirer sets wantToDeprovision on the host to be parked if we already have >=20 hosts parked by Agent.ProvisionedExpirer